xds/src/test/java/io/grpc/xds/XdsSecurityClientServerTest.java

```diff
@@ -103,10 +113,13 @@ public class XdsSecurityClientServerTest {
   private static final String OVERRIDE_AUTHORITY = "foo.test.google.fr";
 
   @After
-  public void tearDown() {
+  public void tearDown() throws IOException {
     if (fakeNameResolverFactory != null) {
       NameResolverRegistry.getDefaultRegistry().deregister(fakeNameResolverFactory);
     }
+    if (testName.getMethodName().equals("tlsClientServer_useSystemRootCerts")) {
```

Comment:
Please no. This is unlikely to be noticed, is trivial to break, and is a huge code smell. Set a File field to non-null and check it here, use try-catch to delete it in the test, TemporaryFolder, or `File.deleteOnExit()`.

Reply:
Done.